xknx/devices/fan.py

```diff
@@ -58,11 +60,29 @@ def __init__(
         self.mode = FanSpeedMode.STEP if max_step else FanSpeedMode.PERCENT
         self.max_step = max_step
 
+        # If there is no dedicated speed GA, then the main GA of the fan serves
+        # as speed GA (and implicitely on/off as well). If however there is a dedicated GA,
+        # then the regular state GA becomes an on/off switch.
+        if group_address_speed is not None:
+            self.switch = RemoteValueSwitch(
```

Comment:
just initialize it with `None` group addresses. Otherwise things like `return self.switch.initialized` would raise `AttributeError`.

Reply:
Change pushed